server/services/model.service.js

```diff
@@ -0,0 +1,79 @@
+import mongoose from 'mongoose';
+import { ErrorHandler } from '../helpers/errorHandler';
+import httpStatus from '../helpers/httpStatus';
+import { USER_ROLE, VISITATION_STATUS } from '../helpers/constants';
+import Enquiry from '../models/enquiry.model';
+import Offer from '../models/offer.model';
+import OfflinePayment from '../models/offlinePayment.model';
+import Property from '../models/property.model';
+import Referral from '../models/referral.model';
+import ReportedProperty from '../models/reportedProperty.model';
+import Transaction from '../models/transaction.model';
+import User from '../models/user.model';
+import Visitation from '../models/visitation.model';
+
+const { ObjectId } = mongoose.Types.ObjectId;
+
+const getAllModels = async (user) => {
+  let matchObject;
```

Comment:
Instead of defining the empty objects in the `else` block, I think it is better you define them here so that we can remove the else block.

```suggestion
  let matchObject = {};
```